redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Node.kt

```diff
@@ -0,0 +1,198 @@
+
+  /**
+   * The flex grow attribute of the node.
+   *
+   * The attribute determines how much this child will grow if positive free space is
+   * distributed relative to the rest of other nodes included in the same flex line.
+   * If not specified, [DefaultFlexGrow] is set as a default value.
+   */
+  public val flexGrow: Float
+
+  /**
+   * The flex shrink attribute of the node.
+   *
+   * The attribute determines how much this child will shrink if negative free space is
+   * distributed relative to the rest of other nodes included in the same flex line.
+   * If not specified, [DefaultFlexShrink] is set as a default value.
+   */
+  public val flexShrink: Float
+
+  /**
+   * The flexBasisPercent attribute of the node.
+   *
+   * The attribute determines the initial node length in a fraction format relative to its
+   * parent.
+   * The initial main size of this child View is trying to be expanded as the specified
+   * fraction against the parent main size.
+   * If this value is set, the length specified from layout_width
+   * (or layout_height) is overridden by the calculated value from this attribute.
+   * This attribute is only effective when the parent's MeasureSpec mode is
+   * MeasureSpec.EXACTLY. The default value is -1, which means not set.
+   */
+  public val flexBasisPercent: Float
+
+  /**
+   * The align self attribute of the node.
+   *
+   * The attribute determines the alignment along the cross axis (perpendicular to the
+   * main axis). The alignment in the same direction can be determined by the
+   * align items attribute in the parent, but if this is set to other than
+   * [AlignSelf.Auto], the cross axis alignment is overridden for this child.
+   * The value needs to be one of the values in ([AlignSelf.Auto],
+   * [AlignItems.Stretch], [AlignItems.FlexStart], [AlignItems.FlexEnd],
+   * [AlignItems.Center], or [AlignItems.Baseline]).
+   * If not specified, [AlignSelf.Auto] is set as a default value.
+   */
+  public val alignSelf: AlignSelf
+
+  /**
+   * The wrapBefore attribute of the node.
+   *
+   * The attribute forces a flex line wrapping. i.e. if this is set to `true` for a
+   * node, the item will become the first item of the new flex line. (A wrapping happens
+   * regardless of the nodes being processed in the the previous flex line)
+   * This attribute is ignored if the flex_wrap attribute is set as nowrap.
+   * The equivalent attribute isn't defined in the original CSS Flexible Box Module
+   * specification, but having this attribute is useful for Android developers to flatten
+   * the layouts when building a grid like layout or for a situation where developers want
+   * to put a new flex line to make a semantic difference from the previous one, etc.
+   */
+  public val wrapBefore: Boolean
+
+  /**
+   * The margin of the node.
+   */
+  public val margin: Spacing
+
+  /**
+   * The measured width after invoking [measure].
+   */
+  public val measuredWidth: Int
```

Comment:
Add docs to describe what happens if I invoke this before calling measure()?

Reply:
One of the major challenges of this API is the mix of in and out params.

Reply:
+1 I'd like to have `measure` return a `Size` (like `FlexboxEngine.measure`) instead of holding onto some mutable state, but it was a bigger change to the code and I want to ensure I have good regression tests set up before making that change.